app/apphandlers/embeddedAppHandler.go

```diff
@@ -189,6 +192,18 @@ func (a *embeddedApp) StartRudderCore(ctx context.Context, options *app.Options)
 		}))
 	}
 
+	var eventSchemasJobsDB *jobsdb.HandleT
+	if a.config.enableEventSchemasJobsDB {
+		eventSchemasJobsDB = jobsdb.NewForWrite(
+			"event_schemas",
+			jobsdb.WithClearDB(options.ClearDB),
+			// jobsdb.WithStatusHandler(),
+			jobsdb.WithDSLimit(&a.config.processorDSLimit),
+		)
+	} else {
+		eventSchemasJobsDB = jobsdb.NewForNoop("event_schemas")
```

Comment:
for convenience reasons:
because of several issues I was facing with the `*HandleT` type and `jobsdb.JobsDB` interface type

Reply:
a simple `proc.schemaDB != nil` doesn't work
`proc.schemaDB.(*jobsdb.HandleT) != nil` results in:
`interface conversion: jobsdb.JobsDB is *mocks_jobsdb.MockJobsDB, not *jobsdb.HandleT`
having a `noop` jobsdb that doesn't run any routines in the background was cheap and made life convenient.

Reply:
tangentially, what was the reason to have a `JobsDB` interface defined in the jobsdb package..?
Why not have processor take something like this for the jobsdb:
```
func (proc *Handle) Setup(
	backendConfig backendconfig.BackendConfig,
	gatewayDB Reader,
	routerDB, batchRouterDB, errorDB, schemaDB Writer...)


type Reader interface {  // or any appropriate name considering we also update status
	GetUnprocessed
	UpdateJobStatus
}

type Writer interface {
	Store
	StoreSafe
	.
	.
	.
}
```

instead of like now
```
unc (proc *Handle) Setup(
	backendConfig backendconfig.BackendConfig,
        gatewayDB, routerDB, batchRouterDB, errorDB, schemaDB jobsdb.JobsDB,
```

Reply:
I am not convinced that a noop jobsdb is the best way forward.

Reply:
I'm open to ideas.
with the approach you suggested above(check if nil) I'm facing the mentioned issues atm.